builtin/providers/aws/resource_aws_lambda_function.go

```diff
@@ -237,7 +237,7 @@ func resourceAwsLambdaFunctionRead(d *schema.ResourceData, meta interface{}) err
 
 	getFunctionOutput, err := conn.GetFunction(params)
 	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" && !d.IsNewResource() {
```

Comment:
I know it's unrelated to the original intention behind this PR, but it would be great to have a `[WARN]` log message here, like we have in most other resources. 😉